updated dental table logic

Rohithone opened the pull request:
https://www.pivotaltracker.com/n/projects/2481184/stories/184416351

nisanthyaganti9 reviewed (changes requested):
Rohith, as discussed these changes can be reverted. We only need to consider dental reference plan when calculating costs.

update [this](https://github.com/health-connector/enroll/blob/feature/2481184-employee-detailed-costs/components/sponsored_benefits/app/models/sponsored_benefits/services/plan_cost_service.rb#L78) with below code -

```
plans = [@benefit_group.dental_reference_plan]
```

Rohithone commented:
> Rohith, as discussed these changes can be reverted. We only need to consider dental reference plan when calculating costs.
> 
> update [this](https://github.com/health-connector/enroll/blob/feature/2481184-employee-detailed-costs/components/sponsored_benefits/app/models/sponsored_benefits/services/plan_cost_service.rb#L78) with below code -
> 
> ```
> plans = [@benefit_group.dental_reference_plan]
> ```

@nisanthyaganti9 I reverted the changes and updated plan_cost_service.rb, please check and review.
